src/controller_examples/rabbitmq_controller/proof/liveness/resource_match.rs

```diff
@@ -192,97 +212,79 @@ proof fn lemma_from_after_get_resource_step_and_key_not_exists_to_resource_match
+                            assert(s_prime.ongoing_reconciles()[rabbitmq.object_ref()].local_state.reconcile_step == after_get_k_request_step(next_resource));
+                            assert(next_state(s_prime));
+                        } else {
+                            assert(pre(s_prime));
+                        }
+                    }
+                    _ => {
+                        assert(pre(s_prime));
+                    }
                 }
-            },
-            _ => {}
+            }
+            RMQCluster::lemma_pre_leads_to_post_by_controller(
+                spec, (Some(resp_msg), Some(rabbitmq.object_ref())), stronger_next, RMQCluster::continue_reconcile(), pre, next_state
+            );
         }
+        leads_to_exists_intro(spec, known_ok_resp, lift_state(next_state));
+        let exists_ok_resp = lift_state(at_after_create_resource_step_and_exists_ok_resp_in_flight(sub_resource, rabbitmq));
+        assert_by(
+            tla_exists(known_ok_resp) == exists_ok_resp,
+            {
+                assert forall |ex| #[trigger] exists_ok_resp.satisfied_by(ex) implies tla_exists(known_ok_resp).satisfied_by(ex) by {
+                    let resp_msg = choose |resp_msg| {
+                        &&& #[trigger] ex.head().in_flight().contains(resp_msg)
+                        &&& Message::resp_msg_matches_req_msg(resp_msg, ex.head().ongoing_reconciles()[rabbitmq.object_ref()].pending_req_msg.get_Some_0())
+                        &&& resp_msg.content.get_create_response().res.is_Ok()
+                        &&& state_after_create_or_update(sub_resource, resp_msg.content.get_create_response().res.get_Ok_0(), ex.head().ongoing_reconciles()[rabbitmq.object_ref()].local_state).is_Ok()
+                    };
+                    assert(known_ok_resp(resp_msg).satisfied_by(ex));
+                }
+                temp_pred_equality(tla_exists(known_ok_resp), exists_ok_resp);
+            }
+        );
+        valid_implies_implies_leads_to(spec, match_and_ok_resp, exists_ok_resp);
+        leads_to_trans_n!(spec, pre, match_and_ok_resp, exists_ok_resp, lift_state(next_state));
     }
-
-    assert forall |s, s_prime| pre(s) && #[trigger] stronger_next(s, s_prime) && RMQCluster::kubernetes_api_next().forward(input)(s, s_prime)
-    implies post(s_prime) by {
-        let resp_msg = RMQCluster::handle_get_request(req_msg, s.kubernetes_api_state).1;
-        assert({
-            &&& s_prime.in_flight().contains(resp_msg)
-            &&& Message::resp_msg_matches_req_msg(resp_msg, req_msg)
-            &&& resp_msg.content.get_get_response().res.is_Err()
-            &&& resp_msg.content.get_get_response().res.get_Err_0().is_ObjectNotFound()
-        });
-    }
-
-    RMQCluster::lemma_pre_leads_to_post_by_kubernetes_api(
-        spec, input, stronger_next, RMQCluster::handle_request(), pre, post
-    );
 }
 
 proof fn lemma_from_after_get_resource_step_and_key_exists_to_resource_matches(
```

Comment:
Can you add more comments to this lemma? Briefly explain each of your high-level proof step below